### DISCUSSION TITLE
x25519 key usage

### DISCUSSION THREAD

**Discussion by dev-area** [polhenarejos/pico-hsm#51] [Q&A] [answered]
hi
i built the new 4.2-eddsa1 and loaded it to the board

next , using pico-hsm-tool i initialized and generated x25519 key pair:

$ python3 pico-hsm-tool.py initialize
$ python3 pico-hsm-tool.py keygen  x25519

Pico HSM Tool v1.14
Author: Pol Henarejos
Report bugs to https://github.com/polhenarejos/pico-hsm/issues

Key generated successfully.
Key ID: 1

-----

How can i use this key  (opensc-tool/openssl/pico-hsm-tool/....)  to sign and verify
thanks

**Comment by polhenarejos** [comment, marked as answer]
x25519/x448 are not Edwards curves (Ed25519 or Ed448). 

`pico-hsm-tool` does not provide support for Edwards curves. I'll make a commit.

The way to sign is the same as other keys, nothing special is required.

**Reply by polhenarejos** [reply]
[f450279](https://github.com/polhenarejos/pico-hsm/commit/f4502793706c4e635cec6195de350859b0e3724d) adds support for Edwards curve generation.

**Reply by dev-area** [reply]
thanks!!!
